mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.java

```diff
@@ -288,6 +289,13 @@ protected void updateAndInitPreferences() {
             updateVibrationPreferenceIcon(vibrationPref,
                     prefs.getString(Constants.PREFERENCE_NOTIFICATION_VIBRATION, ""));
 
+            demoModePref.setOnPreferenceChangeListener((preference, newValue) -> {
+                if ((boolean) newValue) {
+                    scheduleWorker(preference.getKey());
```

Comment:
... call `scheduleWorker(Constants.PREFERENCE_ALARM_CLOCK)` here irregardless of the new pref value.